academic-observatory-workflows/academic_observatory_workflows/openalex_telescope/telescope.py

```diff
@@ -0,0 +1,502 @@
+                gke_conn_id=gke_conn_id,
+            )
+            for key in gke_volume_size_map
+        }
+
+
+def create_dag(dag_params: DagParams) -> DAG:
+    @dag(
+        dag_id=dag_params.dag_id,
+        start_date=dag_params.start_date,
+        schedule=dag_params.schedule,
+        catchup=False,
+        max_active_runs=dag_params.max_active_runs,
+        tags=["academic-observatory"],
+        default_args={
+            "owner": "airflow",
+            "on_failure_callback": on_failure_callback,
+            "retries": dag_params.retries,
+            "queue": dag_params.queue,
+        },
+    )
+    def openalex():
+        @task(multiple_outputs=True)
+        def fetch_entities(dag_params: DagParams, **context) -> dict:
+            """Fetch OpenAlex releases.
+
+            :return: True to continue, False to skip.
+            """
+
+            import academic_observatory_workflows.openalex_telescope.tasks as tasks
+
+            dag_run = context["dag_run"]
+            is_first_run = is_first_dag_run(dag_run)
+            entity_index = tasks.fetch_entities(
+                dag_id=dag_params.dag_id,
+                run_id=context["run_id"],
+                is_first_run=is_first_run,
+                entity_names=dag_params.entity_names,
+                cloud_workspace=dag_params.cloud_workspace,
+                schema_folder=dag_params.schema_folder,
+                bq_dataset_id=dag_params.bq_dataset_id,
+                api_bq_dataset_id=dag_params.api_bq_dataset_id,
+                aws_conn_id=dag_params.aws_conn_id,
+                aws_openalex_bucket=dag_params.aws_openalex_bucket,
+            )
+            entity_index_id = release_to_bucket(entity_index, dag_params.cloud_workspace.transform_bucket)
+
+            return {"entity_index_id": entity_index_id, "entity_index": entity_index}
+
+        @task.short_circuit
+        def short_circuit(entity_index: dict, **context):
+            return len(entity_index) > 0
+
+        @task
+        def create_dataset(dag_params: DagParams, *context) -> None:
+            """Create datasets."""
+
+            bq_create_dataset(
+                project_id=dag_params.cloud_workspace.output_project_id,
+                dataset_id=dag_params.bq_dataset_id,
+                location=dag_params.cloud_workspace.data_location,
+                description=dag_params.dataset_description,
+            )
+
+        @task_group
+        def process_entity(
+            entity_index_id: str,
+            entity_index: dict,
+            entity_name: str,
+            dag_params: DagParams,
+            gke_params: GkeParams,
+        ):
+            @task()
+            def aws_to_gcs_transfer(entity_index: dict, entity_name: str, dag_params: DagParams, **context):
+                """Transfer files from AWS bucket to Google Cloud bucket"""
+
+                import academic_observatory_workflows.openalex_telescope.tasks as tasks
```

Comment:
you can just import the tasks globally and call them as you normally would for the non-k8s tasks

Reply:
I was thinking about this as I wrote the workflow, and I think that you were right for doing this originally for workflows with mixed kubernetes and normal tasks, because otherwise you don't get a warning that you haven't imported `tasks` into the kubernetes decorated function.

Reply:
Ah that's a good point!